src/textual/widget.py

```diff
@@ -1049,6 +1049,7 @@ def on_idle(self, event: events.Idle) -> None:
             self.screen.post_message_no_wait(messages.Layout(self))
         self._layout_required = False
         self._repaint_required = False
+        await self.invoke_and_clear_callbacks()
```

Comment:
Sorry, it occurs to me that the screen won't be refreshed at this point. A message has been sent to `screen` but the actual update won't occur until screen idle.

I think the actual redraw occurs in `_on_update`